src/main/java/frc/robot/subsystems/Climber.java

```diff
@@ -0,0 +1,118 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMax.ControlType;
+import com.revrobotics.CANSparkMax.SoftLimitDirection;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import static frc.robot.Constants.CAN;
+import static frc.robot.Constants.ClimbSettings;
+
+public class Climber {
+    // PIDSlot used
+    int slot = 0;
+
+    // raise/lower controllers
+    private CANSparkMax left_extender= new CANSparkMax(CAN.CMB_LEFT_Extend, MotorType.kBrushless);
+    private CANSparkMax right_extender = new CANSparkMax(CAN.CMB_RIGHT_Extend, MotorType.kBrushless);
+
+    // rotation arm controller (outer arms rotate)
+    // private CANSparkMax l_rotator = new CANSparkMax(CAN.CMB_L_Rotate, MotorType.kBrushless);
+    // private CANSparkMax r_rotator = new CANSparkMax(CAN.CMB_R_Rotate, MotorType.kBrushless);
+
+    public Climber() {
+        // TODO - raise/lower are actions ie commands. In the sub-sys we are defining
+        // devices
+        // and behaviors. Left/Right Inner/Outer arms left-rotator, right-rotator
+        // gives a better idea of what
+
+        // TODO Set the control type and PID settings - most likely position control
+
+        // Copy the PID settings down to the SparkMax hardware 
+        // Rotation PIDS
+        // ClimbSettings.armPID.copyTo(l_rotator.getPIDController(), slot);
+        // ClimbSettings.armPID.copyTo(r_rotator.getPIDController(), slot);
+        // // Arm extension PIDS
+        ClimbSettings.innerPID.copyTo(left_extender.getPIDController(), slot);
+        ClimbSettings.innerPID.copyTo(right_extender.getPIDController(), slot);
+
+
+    }
+
+
+    public boolean readyToClimb() {
+        // returns if robot is in the right position, and all motors are in place
+        return false;
+    }
+
+    public String rung() {
+        return "blank";
+    }
+
+    public boolean readyToTraverse() {
+        return false;
+    }
+    // @param inches from extender current position
+    public void setExtension(double inches) {
+        float count =  (float) inches; // convert command inches to encoder counts
+        SoftLimitDirection direction = null ;
+        double speed = 0 ;
```

Comment:
Rather then leaving this in a open loop control, we prefer to use PID loops and closed control systems running on the actual device for a more fine grained control.

For a example of a position PID loop see this [demo project](https://github.com/REVrobotics/SPARK-MAX-Examples/tree/master/Java/Position%20Closed%20Loop%20Control) by REV. Specifically this [line](https://github.com/REVrobotics/SPARK-MAX-Examples/blob/c18eb6f15f9d9b8d371da83e7302470f0d50c170/Java/Position%20Closed%20Loop%20Control/src/main/java/frc/robot/Robot.java#L112).

Reply:
@dlaufen Could you link some example code that shows how to change from native units (rotations/pulses?) to sane (inches in this case). Would be a nice complement to add while making this change